### ISSUE TITLE
Xkcd

### ISSUE THREAD

**Issue by fazo96** [ipfs/archives#21]
I plan to archive all the comics in http://xkcd.com/

I think i'll use (comicnumber).png for the image and (comicnumber).txt to store the alt-text (it often contains jokes)

**Comment by whyrusleeping** [comment]
@fazo96 how are you going to manage the comics that are dynamic or contain multiple sequential images? Or the map ones that have a larger version available on click?

I would *loooooove* to have this. But we should make sure that randall is okay with it first, i'm not sure if there are any sort of copyrights involved here. (too bad he doesnt use github, or we could just ping him)

**Comment by jbenet** [comment]
> I would loooooove to have this. But we should make sure that randall is okay with it first, i'm not sure if there are any sort of copyrights involved here.

absolutely. thanks for saying this.

looks like everything is released CC-BY-NC 

> (too bad he doesnt use github, or we could just ping him)

i'm sure has an account. just have to find it \o/

@fazo96 
- would be great to include a web viewer with the archive. 
- maybe make a dir for every comic
- put the image in both `image.png` and `<original-img-filename>` so that we respect his filenames too, but also make them predictably linked
- put the alt text in a file, like `alt.txt`


(Alternatively, mirror the RSS feed?)

**Comment by rnhmjoj** [comment]
The title and the alt text could be stored in the png metadata. You can use ImageMagic: see [this](http://blog.client9.com/2007/08/19/editing-png-metadata-from-command-line.html).

**Comment by fazo96** [comment]
Looks like license is not an issue as long as we provide credit to randall and include a copy of the license.

Storing title and alt-text in the `png` metadata looks like the way to go!

As far as unconventional comics, we'll figure a solution out for every comic

**Comment by davidar** [comment]
:+1: 

> a viewer would be great, but at this point, what do you guys think about including the entire website?

I think #7 is also quite relevant here.

**Comment by fazo96** [comment]
I completed the archive (now every image file and more is available via ipfs), it just a needs a viewer and probably better folder structure.

Here you go: [QmPVP4sDre9rtYahGvcjv3Fqet3oQyqrH5xS33d4YBVFme](http://gateway.ipfs.io/ipfs/QmPVP4sDre9rtYahGvcjv3Fqet3oQyqrH5xS33d4YBVFme)

**Comment by davidar** [comment]
@fazo96 :clap:

**Comment by cryptix** [comment]
@fazo96 :+1: 

Can we zero pad the numbers on the next pass? :)

**Comment by fazo96** [comment]
@cryptix yeah I figured it was necessary :) if you'd like a try, the script I used to generate the directory tree is included in the directory. It's named `xkcd-downloader.js`

If I have time I'll implement it

**Comment by mateon1** [comment]
I have scraped the entirety of xkcd.com and some of it's subdomains, the result is a very well functioning copy, available here: https://ipfs.io/ipfs/QmbcvivamWCKUuQjdTbCHNBy74qehU6uWTCdyBw3sN8X6b

**Comment by Kubuxu** [comment]
Looks like the currently referenced version on the website isn't fully available.

**Comment by leerspace** [comment]
@fazo96 do you have the original archive that's currently linked to on the archives.ipfs.io site? https://ipfs.io/ipfs/QmPVP4sDre9rtYahGvcjv3Fqet3oQyqrH5xS33d4YBVFme

It doesn't currently seem to be fully available, but if you still have it I can pin it to my ipfs node. I'd try to reproduce the archive using the script in the archive, but I could only guess what the exact text was in the about and license files.

**Comment by leerspace** [comment]
FWIW I just generated a new version of fazo96's archive that's linked to from the site and pinned it to my ipfs node, so the comics that I couldn't access through the gateway before now seem to be accessible. The `about` and `license` files still seem to be unavailable, but I just added the relevant pages from the website to this version of the archive.

[Qmb8wsGZNXt5VXZh1pEmYynjB6Euqpq3HYyeAdw2vScTkQ](https://ipfs.io/ipfs/Qmb8wsGZNXt5VXZh1pEmYynjB6Euqpq3HYyeAdw2vScTkQ)

**Comment by lgierth** [comment]
Awesome, gonna pull that onto one of our storage nodes too. @leerspace wanna make a PR to update the site?

**Comment by lgierth** [comment]
Cool thanks, I just update https://archives.ipfs.io

**Comment by fazo96** [comment]
@leerspace sorry for replying late, looks like I lost my copy of the original archive. Thanks for updating it! :+1:

**Comment by chosenken** [comment]
Hello, I've updated the archive using the `xkcd-downloader.js` script offered in the repo, and it now has all comics up to the latest today (1862).  It is currently pinned on my laptop, but I will pin it to my server when I get home so it will be available at all times.

[QmdmQXB2mzChmMeKY47C43LxUdg1NDJ5MWcKMKxDu7RgQm](https://ipfs.io/ipfs/QmdmQXB2mzChmMeKY47C43LxUdg1NDJ5MWcKMKxDu7RgQm)

**Comment by lgierth** [comment]
Awesome, thanks @chosenken -- also pinned it on nihal.i.ipfs.io

**Comment by chosenken** [comment]
Updated again to 1864, but this time attached it to an ipns: `QmTaW8vRj4SkM6JhqVhAsibQE9PdJb5PQ2FMwPPc6gBi2h`.  I might work on a script that pulls new comics down and updates the ipns when it changes.

**Comment by carsonfarmer** [comment]
I'd like to update this one again, but to facilitate programmatic access, I'd like to change the structure slightly to something more like:
```
/ipfs/Qmahash/1/1 - Barrel - Part 1.png
...
/ipfs/Qmbhash/2003/2003 - Presidential Succession.png
```
where the comic files are contained within a 'folder' defined by the number rather than number and name.

**Comment by olizilla** [comment]
@carsonfarmer that'd be rad. I've no objection to simplifying the folder structure.

**Comment by olizilla** [comment]
I plan to feature this data set on the start page of the new IPLD Explorer page in the ipfs-webui.

**Comment by olizilla** [comment]
@carsonfarmer could we get some zero padding on those indexes?

```
/ipfs/QmHash/0001/0001 - Barrel - Part 1.png
...
/ipfs/Qmbhash/2003/2003 - Presidential Succession.png
```

**Comment by carsonfarmer** [comment]
Ah sorry, was on vacation. Yes I'll update the indexes and post here when ready.

**Comment by HugoReeves** [comment]
I've written a new program in go that creates an archive such as the following, [/ipfs/Qmc7D35JiGQXvRS9HWxdZoa97V3bQqyiekRWmnCgS7RURi](https://ipfs.io/ipfs/Qmc7D35JiGQXvRS9HWxdZoa97V3bQqyiekRWmnCgS7RURi). It also includes a csv with an index of post titles, published date and post number. I have submitted a pr, ipfs/awesome-ipfs#193

**Comment by hodlon** [comment]
If I were someone who wanted to start pinning content like XKCD their own node(s) to help network redundancy, which hash would I use? There are many different hashes presented in this thread and it's not clear to me which one is the most relevant or up-to-date.

This _seems_ to be the most up-to-date: Qmb8wsGZNXt5VXZh1pEmYynjB6Euqpq3HYyeAdw2vScTkQ

This one seems to be an exact replica of the first but it has a different hash, perhaps due to the lack up .html on "about" and "license": QmPVP4sDre9rtYahGvcjv3Fqet3oQyqrH5xS33d4YBVFme

https://archives.ipfs.io/ seems to favor the first one, TkQ.

And this seems to be the same content as everything above but in a different structure: https://ipfs.io/ipfs/QmdAChzF2JQCx9icrmYHZhFdRSv9TpRjq5q1v5b3ANpxRf


So which hash do I pin?

**Comment by rnhmjoj** [comment]
I think it would be a good idea to zero pad the numbers: right know it's impossible to browse.

**Comment by j2ghz** [comment]
I added padding to the downloader: /ipfs/QmX4pR3KKdivwY9Pn5mHNYi5FRhtTqapMfFmW4SYesstxU/xkcd-downloader.js
Whole: https://ipfs.io/ipfs/QmX4pR3KKdivwY9Pn5mHNYi5FRhtTqapMfFmW4SYesstxU
I'm not sure why it's not sorted though

**Comment by hacdias** [comment]
I've just created an XKCD archive at [`/ipns/xkcd.hacdias.com`](https://ipfs.io/ipns/xkcd.hacdias.com/). It is updated every day 😄 Please see the repository for more info: https://github.com/hacdias/xkcd.hacdias.com

**Comment by hacdias** [comment]
@Stebalien should we update the index to `/ipns/xkcd.hacdias.com`?

**Comment by Stebalien** [comment]
I take it your comment is moot now that this has all moved to awesome.ipfs.io?

**Comment by Stebalien** [comment]
Is it possible to have _two_ links? Ideally, we'd link to an immutable version as well.

**Comment by hacdias** [comment]
@Stebalien we could also add an immutable version, but that would be a snapshot somewhere in the past. But yes, we could add it as a description perhaps.

**Comment by Stebalien** [comment]
Yeah, I know. It's just that archives that rely on DNS sketch me out a bit.

**Comment by hacdias** [comment]
Does this look good:

![image](https://user-images.githubusercontent.com/5447088/60753060-83caf200-9fc5-11e9-9ae2-bc309d025254.png)

or some history/clock-like icon near the bottom buttons'd look better?
